### DISCUSSION TITLE
Test.Platform how to add dotnet test parameter when invoke through dotnet test.dll

### DISCUSSION THREAD

**Discussion by jundayin** [microsoft/testfx#4085] [Q&A] [answered]
Hi, 

I just checked the video of Test.Platform and did some PoC on my project. Feels great, I can feel a lot of performance boost when running some heavy E2E API tests.

However, when I tried to run it inside the container and invoke the test project by dll directly instead of by `dotnet test`. I don't see there is an option for me to configure the tests to be included in the running as `dotnet test --filter "xxxx"` providing. Is it even supported?

**Comment by Evangelink** [comment]
Here is our documentation https://learn.microsoft.com/dotnet/core/testing/unit-testing-platform-integration-dotnet-test#dotnet-test---microsofttestingplatform-mode. The easiest in my opinion is the extra `--` so you would do `dotnet test -- --filter "xxx"`. 

Note that we are working on a new integration in dotnet test that would simplify parameter handling but that's wip at the moment. We will do a blogpost when available.

**Reply by jundayin** [reply]
@Evangelink thanks for the answer. But as far as I understand, if I still want to invoke through `dotnet test` then my docker image should still include the dotnet SDK instead of just runtime. Is that correct?

**Reply by nohwnd** [reply, marked as answer]
In docker you should run the executable / dll directly. Either by executing it, or via dotnet exec. You can then specify --filter to filter the tests (or --help to print all the options available).

Showing on windows but it is the same principle. 

![image](https://github.com/user-attachments/assets/1c1fc8b6-e1a3-4b65-a5bb-1457d5e34701)

In this case you don't need SDK in the image.

**Reply by nohwnd** [reply]
You might also see error about the location not being writeable, testing platform is trying to create a directory next to the executable, which is problematic in docker, you might need to override the test result folder to a writeable location by `--results-directory` or setting `platformOptions__resultDirectory` env variable in your image to set this option. Please notice that there are 2 underscores, not just 1.

**Reply by Evangelink** [reply]
To add up, yes if you want to run `dotnet test` you will need the SDK installed because that's a gesture coming from the SDK.
Hopefully, calling the exe directly is not a problem for you.

PS: If you don't mind, I would be super interested about the perf boost you are seeing after moving to the new platform!

**Reply by nohwnd** [reply]
Once you are done with it, would it be possible to see your solution? I think it would be nice to write down the approaches into a doc.
